src/random_ray/random_ray_simulation.cpp

```diff
@@ -314,18 +413,20 @@ void RandomRaySimulation::simulate()
     }
 
     // Execute all tallying tasks, if this is an active batch
-    if (simulation::current_batch > settings::n_inactive && mpi::master) {
-
-      // Generate mapping between source regions and tallies
-      if (!domain_->mapped_all_tallies_) {
-        domain_->convert_source_regions_to_tallies();
-      }
-
-      // Use above mapping to contribute FSR flux data to appropriate tallies
-      domain_->random_ray_tally();
+    if (simulation::current_batch > settings::n_inactive) {
 
       // Add this iteration's scalar flux estimate to final accumulated estimate
       domain_->accumulate_iteration_flux();
```

Comment:
We are now accumulating the iteration flux more efficiently, as there already exists OMP paralleism in the `accumulate_iteration_flux()` function, but before we were only calling it on the master process, correct?

Reply:
Yes, previously we'd only accumulate to the final flux solution vector on the master rank, as this vector was only ever used for plotting purposes by the master rank. As now all ranks will calculate the adjoint source (which is derived from this vector), all ranks are now making the accumulation as well. In practice, it shouldn't have any effect on overall walltime of the simulation, as previously just the master rank was performing this accumulation (via OpenMP) while the other ranks sat idle. Now, instead of sitting idle, the other ranks will be doing the same work the master is.